### ISSUE TITLE
Negative Sampling Skip Gram model without one hot vector input

### ISSUE THREAD

**Issue by rubenohayon** [fchollet/keras#6961]
I have pairs of movie witch contains 2783 features. 
The vector is defined as: if the feature is in the movie it's 1 otherwise it's 0.
Example : 

`movie 1 = [0,0,1,0,1,0,1 ...] & movie 2 = [1,0,1,1,1,0,1 ...]
`
Each pair has for label 1 or 0.

```
movie1,movie2=0
movie1,movie4=1
movie2,movie150=0
```

The input is similar to SGNS (Skip gram negative sampling) word2vec model.
My goal is to find similarity between programs and learn embedding of each movie. 
I'd to build a kind of 'SGNS implementation with keras'. However my input is not one hot and I can't use the Embedding layers. I tried to use Dense layers and merge them with a dot product. I'm not sure about the model architecture and I got errors.

```
from keras.layers import Dense,Input,LSTM,Reshape
from keras.models import Model,Sequential

n_of_features = 2783
n_embed_dims = 20

# movie1 vectors
word= Sequential()
word.add(Dense(n_embed_dims, input_dim=(n_words,)))

# movie2 vectors
context = Sequential()
context.add(Dense(n_embed_dims, input_dim=n_words,))

model = Sequential()

model.add(keras.layers.dot([word, context], axes=1))

model.add(Dense(1, activation='sigmoid'))

model.compile(optimizer='rmsprop',
			  loss='mean_squared_error')
```

If someone has an idea how to implement it.

**Comment by vladimircape** [comment]
please provide your error

**Comment by rubenohayon** [comment]
https://github.com/fchollet/keras/issues/1787 based on word2vec model. I tried to re create it where input is not one hot

#Model example 

from keras.layers import Dense,Input,LSTM,Reshape,Merge,Activation
from keras.models import Model,Sequential

models = []

# Word vectors
model_word = Sequential()
model_word.add(Dense(300, input_shape=(1,2000)))
model_word.add(Reshape((300,)))
models.append(model_word)

# Context vectors
model_context = Sequential()
model_context.add(Dense(300, input_shape=(1,2000)))
model_context.add(Reshape((300,)))
models.append(model_context)

# Combined model
model = Sequential()
model.add(Merge(models, mode='dot', dot_axes=1))
model.add(Dense(1))
model.add(Activation('sigmoid'))
model.compile(loss='mean_squared_error', optimizer='adam')

**Comment by vladimircape** [comment]
thanks for code, but please provide error logs, they will helps

**Comment by nlothian** [comment]
I think I'm working on a similar problem. There are some clues in this issue, but I haven't worked it out yet: https://github.com/fchollet/keras/issues/3921#issuecomment-308404607
